File: code_snippets/java/src/main/java/usecases/asynctasks/simple/AsyncTaskService.java
package usecases.asynctasks.simple;

import static usecases.utils.ExampleStubs.someHeavyWork;

import dev.restate.sdk.Context;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;
import usecases.utils.TaskOpts;

// <start_here>
@Service
public class AsyncTaskService {

  // <mark_1>
  @Handler
  public String runTask(Context ctx, TaskOpts params) {
    return someHeavyWork(params);
  }
  // </mark_1>
}
// <end_here>

File: code_snippets/kotlin/src/main/kotlin/usecases/asynctasks/simple/AsyncTaskService.kt
package usecases.asynctasks.simple

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.Service
import dev.restate.sdk.kotlin.Context

// <start_here>
@Service
class AsyncTaskService {
  // <mark_1>
  @Handler
  suspend fun runTask(ctx: Context, params: TaskOpts): String {
    return params.someHeavyWork()
  }
  // </mark_1>
}
// <end_here>

class TaskOpts

fun TaskOpts.someHeavyWork(): String {
  return ""
}
